### DISCUSSION TITLE
How to offload processing to the GPU?

### DISCUSSION THREAD

**Discussion by someusernamee** [oobabooga/text-generation-webui#2646] [Q&A]
I'm familiar with GPU layers, but adjusting them in the UI seems to do nothing. There's no increase in VRAM usage or GPU load or anything to indicate that the GPU is being used at all. I'm using Wizard-Vicuna-13B-Uncensored GGML, specifically the q5_K_M version and in the model card it says it's capable of CPU+GPU inferencing with UIs such as oobabooga so I'm not sure what I'm missing or doing wrong here.

**Comment by BetaDoggo** [comment]
If you haven't done so already, llama-cpp-python must be compiled for the webui with cuBLAS or CLBLAST support. The process for that is documented [here](https://github.com/oobabooga/text-generation-webui/blob/main/docs/llama.cpp-models.md). Note that all of the commands have to be run within the webui's environment.
